README.md

```diff
@@ -9,3 +9,664 @@ ExpressRouterAdapter is available as a package on NPM for use with a module bund
+  async getSecurityContext({ req }: any): Promise<ApplicationSecurityContext> {
+    console.log('in here?');
+    const authHeader = req.headers.authorization;
+    // this is NOT real security
+    return new ApplicationSecurityContext(authHeader);
+  }
+
+}
+
+export const getContainer = async (): Promise<Container> => {
+  const container = new Container();
+
+  container.registerAlias(ApplicationRouteProvider, RouteProvider);
+  container.registerAlias(ApplicationSecurityContextProvider, SecurityContextProvider);
+
+  return container;
+};
+```
+
+Now we get the following
+```sh
+curl -H 'authorization:jon' -H 'accept:application/pets+json'  http://localhost:4000/pets/
+[{"firstName":"honey","lastName":"hoguet"},{"firstName":"poseidon","lastName":"hoguet"}]
+```
+
+Which of course isn't secure! But it demonstrates the plumbing.
+
+> Of note: you must have a principal on your security context to not be considered anonymous. You will still get a 401 without a principal property. This should be reconsidered in the next version.
+
+## What if I want a base path added?
+TODO
+
+## What if I need custom timeouts?
+TODO
+
+## How can I provide my own logger?
+TODO
+
+## How do I serialize a response when it isn't json
+TODO
+
+
+## Motivation
+ExpressRouterAdapter was born out of necessity as Symbiotic Labs supported multiple clients building out Restful APIs using Node.
+
+We wanted some basics of REST to be easy / out of the box, so that developers can focus on using REST.
+ - You should just be able to return a model and get a JSON response. You shouldn't have to think about 200, or JSON formatting it.
+ - You should not have to return anything and get 204.
+ - You should be able to use promises / async
+ - You should be able to throw errors, to include errors that can drive status codes
+ - It should be easy to add a new media type to an existing handler (without needing to mess with the content negotiation or thinking about 406 and 415)
+ - You should get a 405 instead of a 404 when hitting a resource that exists but a verb that doesn't
+  > though we haven't gotten to that one yet
+
+We also were concerned about analysis paralysis with all the great things out there (Express, Koa, Strapi, Hapi, Sails, FeatherJS). We are a lean company that favors something that works an delivers business value quickly, over the perfect solution, eventually.
+
+So we decided that something like RouterMetaBuilder would allow us to describe our rest endpoints in a way that could easily be plugged into any framework, and that this would actually give us the most forward portability (over picking the perfect framework). We chose express to the framework we're using because we have experience with it, it is mature, and it has proven a committment to maintaining a solid low level abstraction (as opposed to bloating to be a better abstraction for some use cases at the cost of bloat and being a bad abstraction for other use cases).
```

Comment:
Rephrase `as opposed to bloating to be a better abstraction for some use cases at the cost of bloat and being a bad abstraction for other use cases`

as opposed to trying to handle all possible use cases at the potential cost of increased complexity and code bloat